Control/Monad/Permutations.hs

```diff
@@ -0,0 +1,220 @@
+﻿-- |
+-- Module      :  Control.Monad.Permutations
+-- Copyright   :  © 2017 Mark Karpov
+-- License     :  BSD 3 clause
+--
+-- Maintainer  :  Mark Karpov <markkarpov92@gmail.com>
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- This module is a generalization of the package @parsec-permutation@
+-- authored by Samuel Hoffstaetter:
+--
+-- http://hackage.haskell.org/package/parsec-permutation
+--
+-- This module also takes inspiration from the algorithm is described
+-- in: /Parsing Permutation Phrases/, by Arthur Baars, Andres Loh and
+-- Doaitse Swierstra. Published as a functional pearl at the Haskell
+-- Workshop 2001.
+--
+-- From these two works we derive a flexible and general method for
+-- parsing permutations over an Applicative structure. Quite useful
```

Comment:
Can we quote the word "Applicative" like this `'Applicative'`, then it'll be a link to the `Applicative` type class (here and everywhere).